src/Components/Falling.cs

```diff
@@ -0,0 +1,258 @@
+using System.Collections;
+
+namespace Celeste.Mod.CommunalHelper.Components;
+
+public class Falling : Component
+{
+    // actions
+    public Func<bool> PlayerFallCheck;
+    public Func<bool> PlayerWaitCheck;
+    public Action ShakeSfx;
+    public Action ImpactSfx;
+    public Action LandParticles;
+    public Action FallParticles;
+
+    // config
+    public float FallDelay;
+    public bool ShouldRumble = true;
+    public bool ClimbFall = true;
+    public float FallSpeed = 160f;
+    public bool ShouldManageSafe = true;
+    public bool FallUp = false;
+    public float ShakeTime = 0.4f;
+    public bool EndOnSolidTiles = true;
+    public bool FallImmediately = false;
+
+    // coroutine properties
+    public bool Triggered;
+    public bool HasStartedFalling { get; private set; }
+
+    private Coroutine coroutine;
+
+    public new Solid Entity => EntityAs<Solid>();
+
+    public Falling() : base(false, false)
+    {
+        PlayerFallCheck = OnPlayerFallCheck;
+        PlayerWaitCheck = OnPlayerWaitCheck;
+        ShakeSfx = OnShakeSfx;
+        ImpactSfx = OnImpactSfx;
+        LandParticles = OnLandParticles;
+        FallParticles = OnFallParticles;
+    }
+
+    public override void Added(Entity entity)
+    {
+        // need to call base first to ensure we balance Add/Remove
+        base.Added(entity);
+
+        if (entity is not Solid)
+        {
+            Logger.Log(LogLevel.Warn, "CommunalHelper", $"Attempted to add {nameof(Falling)} to a non-Solid ({entity.GetType().Name})");
```

Comment:
can use `Util.Log` instead so you don't have to specified `"CommunalHelper"`

Reply:
Didn't see that, I just did a search for `Logger.Log` and and found some instances of it, so I used that.